### PR TITLE
Add windows resources to binary

### PR DESCRIPTION
**Pull request by dnephin**
cc @johnstep @jhowardmsft 

I copied a bunch of this over from `moby/moby` and made some minor changes:
* removed the autogen tag and coping of files into a package
* use a temporary directory with trap to remove it
* use `go generate` 

I haven't included any git history, let me know if you think I should

### CODE REVIEW

**Comment by jhowardmsft** [comment]
LGTM (not tested). It would be much nicer if `make.ps1` was ported over to have a native build, not shell-script too....

File: scripts/winresources/event_messages.mc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+MessageId=1
```
Comment (jhowardmsft) [inline scripts/winresources/event_messages.mc]: Actually, I'm not sure these are needed in the CLI. They are for event logs for when the daemon is running as a service.

File: scripts/winresources/resources.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+
+Package winresources is used to embed Windows resources into docker.exe.
+These resources are used to provide
+
+    * Version information
+    * An icon
+    * A Windows manifest declaring Windows version support
+
+The resource object files are generated in hack/make/.go-autogen from
```
Comment (jhowardmsft) [inline scripts/winresources/resources.go]: Paths need updating

Comment (dnephin) [inline scripts/winresources/resources.go]: Oops, I think this file is not used anymore, will remove it.

File: scripts/winresources/event_messages.mc (hunk 1 above)
Comment (dnephin) [inline scripts/winresources/event_messages.mc]: Ok, I will remove this file and the file from the generation script which uses it.

**Comment by dnephin** [comment]
I believe we need this for 17.06 ga, otherwise the binary will be missing the icon and manifest stuff

**Comment by johnstep** [comment]
Yes, agreed. I will review this very soon.

**Pull request merged**